aten/src/ATen/native/native_functions.yaml

```diff
@@ -10425,15 +10425,33 @@
   python_module: linalg
   variants: function
 
-- func: linalg_pinv(Tensor self, float rcond=1e-15, bool hermitian=False) -> Tensor
+- func: linalg_pinv.atol_rtol_tensor(Tensor self, *, Tensor? atol=None, Tensor? rtol=None, bool hermitian=False) -> Tensor
```

Comment:
This is cool but changing the signatures like this is FC-breaking because a torch program that just calls linalg.pinv on a tensor will now serialize this call, and if that program is exported to an older version of PyTorch it won't understand this signature.

We need to add these signatures for 2+ weeks before switching to them as the defaults. (2 weeks is our forward-compatibility window.)

This also wouldn't matter if on one was using linalg.pinv, but it already has adoption, which is cool.

Reply:
Alright, I will investigate what actually happens in this case maybe it doesn't break anything.

Reply:
Let's try serializing these on this build and loading them in PyTorch 1.9 using torchscript and see if they'll still run when `linalg.pinv.(t)` is called.

Reply:
(An alternative may be to make a Python wrapper that dispatches `linalg.pinv(t)` and `linalg.pinv(t, *args, **kwargs)` separately.)

Reply:
Summary:

- `torch.linalg.pinv(x)` _is not_ FC-breaking,
- `torch.linalg.pinv(x, rcond: float)` _is not_ FC-breaking,
- `torch.linalg.pinv(x, hermitian: bool)` _**is**_ FC-breaking.

Honestly, it's still a mystery to me now how this works, but it does and here's how I verified it:

On current PR save .pt file using `torch.jit.script`
```
In [1]: import torch

In [2]: @torch.jit.script
   ...: def test(x):
   ...:     return torch.linalg.pinv(x)
   ...:

In [3]: test.graph
Out[3]:
graph(%x.1 : Tensor):
  %4 : bool = prim::Constant[value=0]()
  %2 : NoneType = prim::Constant()
  %5 : Tensor = aten::linalg_pinv(%x.1, %2, %2, %4) # <ipython-input-2-609569be27ef>:3:11
  return (%5)

In [4]: test.save("test_pinv.pt")
```
Load on other branch ([for example 1.9 in colab](https://colab.research.google.com/drive/1LBIxKZsqcwAcwFiiiIwNTHvAK6FM8h9J?usp=sharing)):
```
In [1]: import torch

In [2]: test = torch.jit.load("test_pinv.pt")

In [3]: test.graph
Out[3]:
graph(%self : __torch__.PlaceholderModule,
      %x.1 : Tensor):
  %6 : bool = prim::Constant[value=0]()
  %5 : float = prim::Constant[value=1.0000000000000001e-15]()
  %7 : Tensor = aten::linalg_pinv(%x.1, %5, %6) # <ipython-input-2-609569be27ef>:3:11
  return (%7)

In [4]: test(torch.randn(3, 3))
Out[4]:
tensor([[ 0.1149,  0.0695, -0.5795],
        [-0.4950, -0.3870, -0.1382],
        [-0.1383,  0.4591,  0.3753]])
```
The interesting thing is that `test.graph` is different. I do not understand yet how `torch.linalg.pinv(x)` and `torch.linalg.pinv(x, hermitian: bool)` are different. @mruberry, do we need to fix `torch.linalg.pinv(x, hermitian: bool)`?